tests/Elasticsearch/Tests/ClientIntegrationTests.php

```diff
@@ -30,8 +54,85 @@ public function testCustomQueryParams()
             'custom' => ['customToken' => 'abc', 'otherToken' => 123],
             'client' => ['ignore' => 400]
         ];
-        $exists = $client->exists($getParams);
+        $exists = self::$client->exists($getParams);
 
         $this->assertFalse($exists);
     }
+
+    /**
+     * @dataProvider aliasDataProvider
+     */
+    public function testExistAlias($alias)
+    {
+        $this->createIndexWithAlias($alias);
+
+        $params = array(
+            'name' => $alias
+        );
+
+        $this->assertTrue(self::$client->indices()->existsAlias($params));
+    }
+
+    /**
+     * @dataProvider aliasDataProvider
+     */
+    public function testFindIndexByAlias($alias)
+    {
+        $this->createIndexWithAlias($alias);
+
+        $params = array(
+            'name' => $alias
+        );
+        $this->assertEquals($alias . '_v1', array_keys(self::$client->indices()->getAlias($params))[0]);
+    }
+
+    /**
+     * @dataProvider aliasDataProvider
+     */
+    public function testPutAlias($alias)
+    {
+
+        $this->createIndexWithoutAlias($alias);
+
+        $params = array(
+            'index' => $alias . '_v1',
+            'name' => $alias
+        );
+
+        self::$client->indices()->putAlias($params);
+
+        $params = array(
+            'name' => $alias
+        );
+        $this->assertEquals($alias . '_v1', array_keys(self::$client->indices()->getAlias($params))[0]);
+    }
+
+    public function aliasDataProvider()
+    {
+        return [
+            'latin-char' => ['myindextest'],
+            'utf-8-char' => ['⿇⽸⾽']
+        ];
+    }
+
+    private function createIndexWithAlias($alias)
```

Comment:
there can be a `string` typehint

Reply:
i'm sorry, i don't understand what do you mean by that :s

Reply:
You can change the method signature to `private function createIndexWithAlias(string $alias)` (added `string`), so the parameter type is checked.

Reply:
thanks for the explanation :)